project-base/storefront/components/Basic/ExtendedNextLink/ExtendedNextLink.tsx

```diff
@@ -37,11 +40,11 @@ export const ExtendedNextLink: FC<ExtendedNextLinkProps> = ({
         if (isWithoutOpeningInNewTab) {
             onClick?.(e);
 
-            if (type) {
-                updatePageLoadingState({ isPageLoading: true, redirectPageType: type });
-            } else {
-                updatePageLoadingState({ redirectPageType: undefined });
-            }
+            const isLinkExternal = isHrefExternal(href, baseUrl);
```

Comment:
How does `isLinkExternal` relate to `isDynamic`. Are they correlated? Will they always be the opposite of each other?

Reply:
**TL;DR** Relying just on the `type` or `isDynamic` doesn't cover all the navigation cases.

Examples explaining their difference:

1. As I have removed the type from the main navigation links - `isDynamic` is false for this case, the same as `isHrefExternal`, because we get the relative pathname, which results in an error and false from the function
2. Imagine someone in the administration putting an absolute URL of the category (eg. `https://ssfwcc.devel.shopsys.cloud/electronics`, which is a valid use case that Vašek Macíček shared with me) - in this case, we don't have a type but we can check that the hostnames match and for this case we can display skeletons.
3. If it is an external link I had an issue in the Arc browser that for the external link the page would open in a new small popup window on top of the current one and the page would be stuck in infinite skeleton loop (similar to the issue with Safari and opening link via mouse middle (wheel) button). In this case, `isDynamic` is false and the condition for `isPageLoading` will result in false too.

Number 3.) is the reason why we cannot simply put `true` to the `isPageLoading`. The `redirectPageType` could be simplified by removing the condition and provide the `DEFAULT_PAGE_TYPE` every time, but since the `isPageLoading` is false anyway it felt like it should be set to `undefined`.

Reply:
Great explanation, thanks.